### ISSUE TITLE
DelayedJob Error Handling Exception

### ISSUE THREAD

**Issue by mike-stewart** [bugsnag/bugsnag-ruby#489]
### Description
The bugsnag gem causes errors when handling DelayedJob errors. This overrides the DelayedJob last_error field, and also what is output in the logs.

### Issue
```
super: no superclass method `error' for #<TestErrorJob:0x007f35c09c7910>
(__DELEGATE__):2:in `error'
bugsnag (6.8.0) lib/bugsnag/integrations/delayed_job.rb:70:in `error'
delayed_job (3.0.3) lib/delayed/backend/base.rb:114:in `call'
delayed_job (3.0.3) lib/delayed/backend/base.rb:114:in `hook'
delayed_job (3.0.3) lib/delayed/backend/base.rb:97:in `rescue in block in invoke_job'
delayed_job (3.0.3) lib/delayed/backend/base.rb:101:in `block in invoke_job'
```

### Environment

Library versions:

```
  * bugsnag (6.8.0)
  * bugsnag-capistrano (2.0.0)
  * delayed_job (3.0.3)
  * delayed_job_active_record (0.3.2)
  * rails (3.2.22.5)
```

- debug mode or production?: Production

### Example Repo

- [ ] Create a minimal repository that can reproduce the issue after running
  `bundle install` and `rackup` (or `rails server`, or `ruby some-script.rb`,
  etc)
- [ ] Link to it here:

**Comment by Cawllec** [comment]
Hi @mike-stewart, have you upgraded any of the above Gems recently, and if so do you have the previous versions you were using?

**Comment by mike-stewart** [comment]
@cawllec I have not upgraded any of the gems recently, except for adding bugsnag.

**Comment by Cawllec** [comment]
How have you installed the plugin within your Rails app?  It shouldn't require any additional setup over what [rails already requires](https://docs.bugsnag.com/platforms/ruby/rails/).

I ask because the error is occurring when the plugin attempts to call `error` function on its `super` class, which should be the `Delayed::Plugins::Plugin` class, and instead appears to be calling a `TestErrorJob` class instead.

**Comment by mike-stewart** [comment]
@Cawllec All I did was add the gem and the initializer as follows:
```ruby
Bugsnag.configure do |config|
  config.api_key = #REDACTED
  config.auto_capture_sessions = true

  config.app_version = File.read("#{Rails.root}/VERSION").chomp

  config.notify_release_stages = ['staging']
end

# Added at the suggestion of Bugsnag support
Bugsnag.before_notify_callbacks << lambda {|report|
  if framework = report.severity_reason[:attributes][:framework]
    report.app_type = framework
  end
}
```

The TestErrorJob is a job that I created to test Bugsnag's DJ integration. It is as follows:
```
class TestErrorJob
  def perform
    raise 'Testing error handling in delayed jobs'
  end
end
```

**Comment by Cawllec** [comment]
Testing against your setup in a clean Rails 3 installation everything seems to work as expected.  Can yo tell me how you're queuing your TestErrorJob, and how you're running the job worker?

**Comment by mike-stewart** [comment]
@Cawllec It's actually breaking for all of our jobs, not just this test job. The jobs get enqueued like `Delayed::Job.enqueue(TestErrorJob.new)` and the workers are run with `bundle exec script/delayed_job`.

**Comment by Cawllec** [comment]
Ok, I'm having a look into it.  We've queued a ticket to update our delayed-job plugin, as the current method we're using may not be ideal, but that won't help your immediate problem.

Would you be able to share your Delayed Job configuration?

**Comment by mike-stewart** [comment]
This is fixed in 6.9.0.
